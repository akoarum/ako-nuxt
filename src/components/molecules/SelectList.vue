<template>
  <ul class="selectList">
    <li v-for="select in selects" :key="select.id" class="selectList__item">
      <ToggleButton v-model="model" :name="name" :value="select.value">
        {{ select.label }}
      </ToggleButton>
    </li>
  </ul>
</template>

<script>
import { ToggleButton } from '~/components/atoms'

export default {
  components: { ToggleButton },
  model: {
    event: 'change'
  },
  props: {
    name: { type: String, required: true },
    selects: { type: Array, required: true },
    value: { type: Array }
  },
  computed: {
    model: {
      set(value) {
        this.$emit('change', value)
      },
      get() {
        return this.value
      }
    }
  }
}
</script>

<style lang="scss" scoped>
.selectList {
  @include media-min {
    display: flex;
    flex-wrap: wrap;
    margin-top: -14px;
    margin-left: percentage(-16px / 1024px);
  }
}

.selectList__item {
  @include media-min {
    margin-top: 14px;
    margin-left: percentage(16px / 1040px);
    width: percentage(244px / 1040px);
  }
}

.selectList__item + .selectList__item {
  @include media-max {
    margin-top: 14px;
  }
}
</style>
