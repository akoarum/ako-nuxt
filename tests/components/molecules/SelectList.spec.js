import { mount } from '@vue/test-utils'
import SelectList from '~/components/molecules/SelectList'

describe('SelectList', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(SelectList, {
      propsData: {
        name: 'test',
        selects: [
          { id: 1, value: '1', label: '項目1' },
          { id: 2, value: '2', label: '項目2' }
        ],
        value: []
      }
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })
    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
