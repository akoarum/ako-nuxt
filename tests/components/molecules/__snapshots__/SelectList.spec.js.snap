// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectList initialized snapshot 1`] = `
<ul
  class="selectList"
>
  <li
    class="selectList__item"
  >
    <label
      class="toggleButton"
    >
      <input
        class="toggleButton__input"
        name="test"
        type="checkbox"
        value="1"
      />
       
      <span
        class="toggleButton__text"
      >
        
      項目1
    
      </span>
    </label>
  </li>
  <li
    class="selectList__item"
  >
    <label
      class="toggleButton"
    >
      <input
        class="toggleButton__input"
        name="test"
        type="checkbox"
        value="2"
      />
       
      <span
        class="toggleButton__text"
      >
        
      項目2
    
      </span>
    </label>
  </li>
</ul>
`;
